File: NativeLibraries/Classica/Classica/Music.m
//
//  Music.m
//  Classica
//
//  Created by Yoshihiro Tanaka on 2017/02/04.
//  Copyright © 2017年 Yoshihiro Tanaka. All rights reserved.
//

#import "Music.h"

@interface Music()

@property (assign, nonatomic, readwrite) NSInteger id;

@property (assign, nonatomic, readwrite) NSInteger composerId;

@property (strong, nonatomic, readwrite) NSString *name;

@property (assign, nonatomic, readwrite) NSInteger year;

@property (strong, nonatomic, readwrite) NSString *opusNumber;

@end

@implementation Music

- (instancetype)initWithId:(NSInteger)id
                composerId: (NSInteger)composerId
                      name:(NSString *)name
                      year:(NSInteger)year
                opusNumber:(NSString *)opusNumber {
    self = [super init];
    if(self)
    {
        self.id = id;
        self.composerId = composerId;
        self.name = name;
        self.year = year;
        self.opusNumber = opusNumber;
    }
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)source {
    self = [super init];
    if(self) {
        self.id = [source[@"id"] intValue];
        self.composerId = [source[@"composerId"] intValue];
        self.name = source[@"name"];
        self.year = [source[@"year"] intValue];
        self.opusNumber = source[@"opusNumber"];
    }
    return self;
}

@end

File: NativeLibraries/Android/Classica/classica/src/main/java/jp/cordea/classica/Music.java
package jp.cordea.classica;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by Yoshihiro Tanaka on 2017/02/13.
 */

public class Music {

    private int id;

    private int composerId;

    private String name;

    private int year;

    private String opusNumber;

    public int getId() {
        return id;
    }

    public int getComposerId() {
        return composerId;
    }

    public String getName() {
        return name;
    }

    public int getYear() {
        return year;
    }

    public String getOpusNumber() {
        return opusNumber;
    }

    public Music(int id, int composerId, String name, int year, String opusNumber) {
        this.id = id;
        this.composerId = composerId;
        this.name = name;
        this.year = year;
        this.opusNumber = opusNumber;
    }

    static Music fromJson(JSONObject jsonObject) {
        try {
            return new Music(
                    jsonObject.getInt("id"),
                    jsonObject.getInt("composerId"),
                    jsonObject.getString("name"),
                    jsonObject.getInt("year"),
                    jsonObject.getString("opusNumber")
            );
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return null;
    }
}
